stempy/reader.cpp

```diff
@@ -23,6 +23,25 @@ using std::vector;
 
 namespace stempy {
 
+Header::Header(uint32_t frameWidth, uint32_t frameHeight,
+               uint32_t imageNumInBlock, uint32_t scanWidth, 
+               uint32_t scanHeight, uint32_t startPosition, vector<uint32_t>& imageNumbers)
+{
+    this->frameWidth = frameWidth;
+    this->frameHeight=frameHeight;
+    this->imagesInBlock = imageNumInBlock;
+    this->scanHeight=scanHeight;
+    this->scanWidth=scanWidth;
+    //if there is no imageNumbers in original data set, reconstruct it
+    if(imageNumbers.size()!=0){
+      this->imageNumbers=imageNumbers;
+
+    }else{
+      //assume the image is consecutive if there is no image numbers included in the data source explicitly
+       throw std::runtime_error("the image_number is supposed to be provided by hdf5 file");
```

Comment:
I don't think we should be thrown an exception here? It probably wants to be from where we construct the Head?